### PR TITLE
Assign cluster score for all underperforming clusters

### PR DESCRIPTION
**Pull request by tataganesh**
## Summary

🎯 **Purpose**

Currently, the underperforming group issue manager assigns a score of less than 1.0 for only those samples that belong to the worst performing cluster. This PR assigns scores to all samples belonging to underperforming clusters. Here, an underperforming cluster is any cluster whose score is less than the mean quality score of the full dataset.

📜 **Example Usage**

Example taken from #1141 

```python
import numpy as np
import pandas as pd
from sklearn.datasets import make_blobs
from sklearn.linear_model import LogisticRegression
from sklearn.model_selection import cross_val_predict
from cleanlab import Datalab

np.random.seed(42)
# Generate synthetic data with blobs
X, y = make_blobs(n_samples=21, centers=3, n_features=2, random_state=42, cluster_std=1.0, shuffle=False)

# Introduce noise to the labels
y[:9] = np.random.randint(0, 2, 9)

# Obtain predicted probabilities using cross-validation
clf = LogisticRegression(random_state=0)
pred_probs = cross_val_predict(clf, X, y, cv=3, method="predict_proba")

# Initialize Datalab and find issues related to underperforming groups
lab = Datalab(data={"X": X, "y": y}, label_name="y", task="classification")
lab.find_issues(
    pred_probs=pred_probs,
    issue_types={
        "underperforming_group": {
            "threshold": 0.75,
            "cluster_ids": np.array([0]*7 + [1]*7 + [2]*7), # Hard-code data slices for testing purposes
        },
    },
)

# Collect and display the identified issues
issues = lab.get_issues("underperforming_group")
results = issues.join(pd.DataFrame({"given_label": y, "predicted_label": pred_probs.argmax(axis=1)}))
print(results)
```

Output

```
Finding underperforming_group issues ...

Audit complete. 7 issues found in the dataset.
    is_underperforming_group_issue  underperforming_group_score  given_label  predicted_label
0                            False                     0.809904            0                0
1                            False                     0.809904            1                0
2                            False                     0.809904            0                0
3                            False                     0.809904            0                0
4                            False                     0.809904            0                0
5                            False                     0.809904            1                0
6                            False                     0.809904            0                0
7                             True                     0.691414            0                1
8                             True                     0.691414            0                1
9                             True                     0.691414            1                0
10                            True                     0.691414            1                0
11                            True                     0.691414            1                0
12                            True                     0.691414            1                1
13                            True                     0.691414            1                1
14                           False                     1.000000            2                2
15                           False                     1.000000            2                2
16                           False                     1.000000            2                2
17                           False                     1.000000            2                2
18                           False                     1.000000            2                2
19                           False                     1.000000            2                2
20                           False                     1.000000            2                2

```

## Impact

🌐 Areas Affected: Underperforming Group Issue Manager
👥 Who’s Affected: Datalab users using the Underperforming Group Issue Manager

## Testing

🔍 Testing Done: A new test was added in `test_datalab.py` to check if samples belonging to different underperforming clusters had a issue score of less than 1.0

 🔗 Test Case Link:  [test_underperforming_clusters_scores](https://github.com/tataganesh/cleanlab/blob/54455afd5ea77729f81458888eb2248db85b7b04/tests/datalab/datalab/test_datalab.py#L1380) 

**Unaddressed Cases**

NA

## Links to Relevant Issues or Conversations

https://github.com/cleanlab/cleanlab/issues/1141

## Reviewer Notes

- I did not update the issue manager tests since the dataset being used has only one label with noise.

### CODE REVIEW

**Review by elisno — APPROVED** [review]
This is great stuff! Thank you for this improvement, @tataganesh!
